### PR TITLE
Allowing Cluster Splitting with HLT Vertices

### PR DESCRIPTION
**Pull request by AdrianoDee**
#### PR description:

This PR proposes the introduction of a new process modifier `hltClusterSplitting` that:
- enables to write the `hltPixelVertices` in `FEVTDEBUGHLTEventContent`;
  - one could test also just the `hltTrimmedPixelVertices` that are just 1-2 for Run3 HLT;
- uses those vertices for the `ak4CaloJetsForTrkPreSplitting` and the cluster splitter;
  -  these producers actually use only the coordinates so the content to be stored could be actually slimmed down to be just a collection of floats;   
- removes the `InitialStepPreSplitting` iteration from iterative tracking.

Running on 14_2_0_pre2 RelVals with Run3 PU:

- [TTbar (400 events)](https://adiflori.web.cern.ch/adiflori/hlt_cluster_splitting/ttbar_plots/);
- [QCD_Pt15To7000_Flat (500 events)](https://adiflori.web.cern.ch/adiflori/hlt_cluster_splitting/qcd_plots/);

the tracking performance seems untouched. 

Screenshot 2024-11-13 alle 16 58 14

I don't know how much this would be useful for Run3, but it may be interesting to check it for Phase2. For Run3 I've measured (roughly)[ a speedup of 1.5-2%](https://adiflori.web.cern.ch/adiflori/hlt_cluster_splitting/circles/web/piechart.php?local=false&dataset=reference_ttbar_pu&resource=time_real&colours=default&groups=packages&show_labels=true&threshold=0) on a TTbar Run3 PU 2025 conditions sample.

I took the chance also to add a specific wf with offset `.19003` (to be aligned to the other cluster splitting related wfs) and a `ProdLike` version with `.1900321`.

#### PR validation:

`runTheMatrix.py -w upgrade -l 17434.19003,17434.1900321`

### CODE REVIEW

**Comment by AdrianoDee** [comment]
type tracking

**Comment by AdrianoDee** [comment]
test parameters:
- relval_opts = -w upgrade
- workflow = 17434.19003

**Comment by AdrianoDee** [comment]
please test

File: RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py [hunk 1]
```diff
@@ -254,6 +262,23 @@
                                        siPixelRecHits,
                                        MeasurementTrackerEvent,
                                        siPixelClusterShapeCache)
+
+hltClusterSplitting.toModify(siPixelClusters,
+    vertices = cms.InputTag("hltPixelVertices")
+    # hltPoint = cms.InputTag("hltTrimmedPixelVertices"),
```
Comment (AdrianoDee) [inline RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py:268]: Also these, will do a cleanup after the tests.

**Comment by antoniovagnerini** [comment]
+1
 - spurious differences seen from unstable WF `12434.7` (open issue #39803 ) 
- spurious differences from HGCal DQM test (open issue  #46416)

**Comment by Moanwar** [comment]
+Upgrade

**Comment by jfernan2** [comment]
@AdrianoDee are you cleaning up the commented out lines finally? Thanks

**Comment by AdrianoDee** [comment]
please test

**Comment by antoniovagnerini** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

**Comment by Moanwar** [comment]
+Upgrade

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by AdrianoDee** [comment]
@cms-sw/dqm-l2 w.r.t. the latest sign-off I only removed a couple of comments. Could you have a look? Thanks ( :

**Comment by antoniovagnerini** [comment]
+dqm

**Comment by mandrenguyen** [comment]
+1

**Comment by mmusich** [comment]
It looks like after this PR was merged several DQM online client related tests started failing, (see e.g. [here](https://cmssdt.cern.ch/SDT/cgi-bin/showBuildLogs.py/el8_amd64_gcc12/www/fri/15.0-fri-23/CMSSW_15_0_X_2024-12-06-2300?utests)). I propose https://github.com/cms-sw/cmssw/pull/46892 to fix it.

**Pull request merged**